README.md

```diff
@@ -1,12 +1,41 @@
-# azure_auth
-
-An example of how to build out Azure SSO authentication and the Javascript renewal of credentials for long lived pages / apps.  This should work in Chrome, Firefox, Edge, Brave and Safari but there is currently no Microsoft support for Opera.
-
-Future Improvements in the lua code:
+Name
+====
+Azure with OpenResty - Example app using openresty, resty-openidc and demonstrating MSAL.js session renewals. 
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Description](#description)
+    * [Services](#services)
+    * [Authentication](#authentication)
+* [Setup](#setup)
+* [Installation](#installation)
+* [Test and Deploy](#test-and-deploy)
+* [Usage](#usage)
+* [Support](#usage)
+* [Contributing](#usage)
+* [Authors and acknowledgment](#authors-and-acknowledgement)
+* [License](#license)
+* [Status](#status)
+
+Description
+===========
+This code demonstrates of how to build out Azure SSO authentication and the Javascript renewal of credentials 
+for long lived pages / apps. This should work in Chrome, Firefox, Edge, Brave and Safari but there is currently 
+no Microsoft support for Opera.  The project will let you setup a basic docker container using environment 
+variables that will be used by Openresty to setup your oidc and then forward it to a django web app and 
+demonstrate JWT token validation.
+
+Future Improvements for the lua code:
 * Ensure the access_token checks id_token.exp, iss, etc
 * Restrict the access_token update to only in keep_alive
+* Demonstrate session refreshing in [resty-openidc](https://github.com/zmartzone/lua-resty-openidc)
+* Show how to do token refreshing with [MSAL.js](https://github.com/AzureAD/microsoft-authentication-library-for-js)
 
-## Web / Azure Keepalive frontend
+
+Services
+--------
 There are two docker containers one respresents a webserver that is around to show how to proxy the connection from auth, display login information, AND show the keep alive method using MSAL.js.
```

Comment:
You mention two containers but seem to only discuss one. What is the other one?

That first sentence is a run on.  

"There are two docker containers. One..."  Alternatively use a colon, but those are so fancy.

Reply:
Modified to describe the containers better.